pkg/sparrow/targets/gitlab.go

```diff
@@ -80,8 +76,8 @@ func (t *gitlabTargetManager) Reconcile(ctx context.Context) error {
 	log := logger.FromContext(ctx)
 	log.Info("Starting global gitlabTargetManager reconciler")
 
-	checkTimer := time.NewTimer(t.checkInterval)
-	registrationTimer := time.NewTimer(t.registrationInterval)
+	checkTimer := time.NewTimer(t.cfg.CheckInterval)
+	registrationTimer := time.NewTimer(t.cfg.RegistrationInterval)
```

Comment:
Can you add a unit test for the 0 case, where the update should never happen? I'm not 100% sure but I think a zero timer returns at least immediately, which would cause the opposite of what we want, 1000s of refreshes every second

Reply:
looked into https://github.com/golang/go/issues/10124, you've already implemented there, very good 👍